AArch64: Initial version of ARM64Instruction and related files

knn-k opened the pull request:
This commit implements the initial version of ARM64Instruction and
related files for aarch64.  This depends on #2660.

Signed-off-by: knn-k <konno@jp.ibm.com>

knn-k commented:
Ready for review.

I am implementing ARM64BinaryEncoding.cpp and ARM64Instructions.cpp, but I am going to put them in another pull request.
In addition, some more ARM64*Instruction classes need to be defined.

knn-k commented on compiler/aarch64/codegen/ARM64ConditionCode.hpp:
```diff
@@ -0,0 +1,51 @@
+   ARMConditionCodeLE,
+   ARMConditionCodeAL,
+   ARMConditionCodeIllegal
+} ARM64ConditionCode;
```
I have my version of compiler/aarch64/codegen/ARM64ConditionCode.hpp in #2704, which is not merged yet.  My version declares all the necessary symbols in its enum.  In addition, the type name is a little bit different: `TR_ARM64ConditionCode` vs `ARM64ConditionCode`.

0xdaryl commented on compiler/aarch64/codegen/GenerateInstructions.cpp:
```diff
@@ -0,0 +1,159 @@
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "compiler/aarch64/codegen/GenerateInstructions.hpp"
```
Does this path need the `compiler/` stem?

knn-k replied:
It came from the p version of GenerateInstructions.cpp.  I am going to change it.

knn-k commented:
I updated GenerateInstructions.cpp reflecting the review comment.
I also moved ARM64ConditionCode.hpp to another PR, #2744, because it conflicts with #2622.

0xdaryl commented:
Merging now because the AppVeyor build has no bearing on these changes.
